mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -550,14 +553,15 @@ class KMeans @Since("1.5.0") (
     blocks.unpersist()
 
     val iterationTimeInSeconds = (System.nanoTime() - iterationStartTime) / 1e9
-    instr.logInfo(f"Iterations took $iterationTimeInSeconds%.3f seconds.")
+    instr.logInfo(log"Iterations took ${MDC(TIME_UNITS, iterationTimeInSeconds%.3f)} seconds.")
```

Comment:
Note: As per https://github.com/apache/spark/pull/45903, we are using ms for all the time related context